paas_app_charmer/_gunicorn/charm.py

```diff
@@ -293,6 +295,7 @@ def _build_charm_state(self) -> CharmState:
             redis_uri=self._redis.url if self._redis is not None else None,
             s3_connection_info=self._s3.get_s3_connection_info() if self._s3 else None,
             saml_relation_data=saml_relation_data,
+            ingress_url=self._ingress.url,
```

Comment:
Won't it be `None` if the ingress relation is not active ?

Reply:
`self._ingress` is always set, `self._ingress.url` will be `None` if the ingress relation is not active. That is, the env variable will not exist in the ingress relation is not active.

Reply:
I'm wondering then if in that case we should fallback to a URL with the service ?
I mean there'll always be a URL since it's a webservice it would make no sense to not have a URL

Reply:
open port 80 and put service name when there is no ingress.